<?xml version="1.0" encoding="utf-8"?>

<launch>
	<arg name="frame_id" default="imu_link_ned"/>
	<arg name="udp_port" default="8002"/>
	<arg name="ip" default="192.168.1.255"/>
	<arg name="time_source" default="ins"/>
	<arg name="time_origin" default="unix"/>
	<arg name="expected_frequency" default="10.0"/>
	<arg name="max_latency" default="1.0"/>
	<arg name="connection_lost_timeout" default="10.0"/>
	<arg name="use_compensated_acceleration" default="false"/>
	<arg name="publish_odom_ins" default="true"/>
	<arg name="dvl_available" default="false"/>

	<node pkg="ixblue_ins_driver" type="node" name="ixblue_ins_driver" output="screen">
		<param name="frame_id" value="$(arg frame_id)"/>
		<param name="udp_port" value="$(arg udp_port)"/>
		<param name="ip" value="$(arg ip)"/>
		<param name="time_source" value="$(arg time_source)"/>
		<param name="time_origin" value="$(arg time_origin)"/>
		<param name="expected_frequency" value="$(arg expected_frequency)"/>
		<param name="max_latency" value="$(arg max_latency)"/>
		<param name="connection_lost_timeout" value="$(arg connection_lost_timeout)"/>
		<param name="use_compensated_acceleration" value="$(arg use_compensated_acceleration)"/>
		<param name="publish_odom_ins" value="$(arg publish_odom_ins)"/>
		<param name="dvl_available" value="$(arg dvl_available)"/>

	</node>

  <!-- Converts Lat/Lon from Microstrain to local X/Y -->
  <node pkg="geonav_transform" type="geonav_transform_node" name="geonav_transform_node" clear_params="true" output="screen" respawn="true">
    <!-- Datum as latitude, longitude [decimal deg.], yaw [ENU, degrees] -->
    <!-- Probably Datum is the offset for latitude and longitude position -->
    <!-- Latitude -12.70715 degrees = -12 degrees and 42.4292 -->
    <!-- Longitude -38.17446 degrees = -38 degrees and 10.4677 -->
    <rosparam param="datum">[-12.70715, -38.17446, 0.0]</rosparam>  
    <param name="base_link_frame_id" value="imu_link_ned" />
    <param name="broadcast_utm2odom_transform" value="false"/>
    <param name="broadcast_odom2base_transform" value="false"/>

    <!-- <param name="odom_frame_id" value="imu_link_ned"/> -->
    <remap from="nav_odom" to="ixblue_ins_driver/ix/ins/odom"/>
  </node>

</launch>
